### PR TITLE
feat: woocommerce dependency check

### PR DESCRIPTION
**Pull request by mastudillot**
This PR adds the functionality to properly handle the WooCommerce dependency in case it is not installed or activated.

Test

Woocommerce is not installed
image

Woocommerce is not activated
image

Webpay transaction ok
image

Oneclick inscription ok
image

Oneclick transaction ok
image

### CODE REVIEW

File: plugin/views/admin/components/notice-missing-woocommerce.php [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<div id="message" class="notice notice-error tbk-notice-container">
+    <div class="tbk-notice">
+        <div>
+            <img src="<?= esc_url($tbkLogo) ?>" height="70px" alt="Transbank logo" />
+        </div>
+
+        <div>
+            <p>WooCommerce no se encuentra activo o no está instalado.</p>
+            <p>
```
Comment (victormendoza96) [inline plugin/views/admin/components/notice-missing-woocommerce.php:9]: unnecessary "p" container

File: plugin/webpay-rest.php [hunk 2]
```diff
@@ -241,3 +220,45 @@ function transbank_rest_check_cancelled_checkout()
...
+            }
+            $actionButton = [];
+
+            if ($isWooInstalled && $currentUserCanInstallPlugins) {
+                $actionButton['text'] = 'Activar Woocoomerce';
+                $actionButton['action'] = esc_html($activateLink);
+            } else {
+                if ($currentUserCanInstallPlugins) {
+                    $actionButton['text'] = 'Instalar Woocoomerce';
+                    $actionButton['action'] = esc_html($installLink);
```
Comment (victormendoza96) [inline plugin/webpay-rest.php:255]:
```
if ($isWooInstalled && $currentUserCanInstallPlugins) {
     $actionButton['text'] = 'Activar Woocoomerce';
     $actionButton['action'] = esc_html($activateLink);
} elseif ($currentUserCanInstallPlugins) {
     $actionButton['text'] = 'Instalar Woocoomerce';
     $actionButton['action'] = esc_html($installLink);
} else {
     $actionButton['text'] = 'Revisar Woocoomerce';
     $actionButton['action'] = 'https://wordpress.org/plugins/woocommerce/';
}
```

**Review by victormendoza96 — APPROVED** [review]

**Review by tribiec — APPROVED** [review]
Besides the @victormendoza96 's comment, it looks great to me!

File: plugin/webpay-rest.php [hunk 3]
```diff
@@ -105,51 +102,60 @@ function( Automattic\WooCommerce\Blocks\Payments\PaymentMethodRegistry $payment_
...
 
-function transbank_webpay_rest_on_webpay_rest_plugin_activation()
-{
-    if (!class_exists('WC_Payment_Gateway')) {
-        trigger_error('Se necesita tener WooCommerce instalado y activo para poder activar este plugin', E_USER_ERROR);
-    }
-    woocommerce_transbank_rest_init();
+function registerPluginActionLinks() {
+    add_filter('plugin_action_links_'.plugin_basename(__FILE__), function ($actionLinks) {
+        $webpaySettingLink = sprintf(
```
Comment (Matiasnickolas) [inline plugin/webpay-rest.php:142]: typo: webpaySettingsLink

File: plugin/webpay-rest.php [hunk 4]
```diff
@@ -105,51 +102,60 @@ function( Automattic\WooCommerce\Blocks\Payments\PaymentMethodRegistry $payment_
...
+            '<a href="%s">%s</a>',
+            admin_url('admin.php?page=wc-settings&tab=checkout&section=transbank_webpay_plus_rest'),
+            'Configurar Webpay Plus'
+        );
+        $oneclickSettingLink = sprintf(
```
Comment (Matiasnickolas) [inline plugin/webpay-rest.php:147]: typo: oneclickSettingsLink

Comment (mastudillot) [reply plugin/webpay-rest.php]: Fixed

File: plugin/webpay-rest.php (hunk 3 above)
Comment (mastudillot) [reply plugin/webpay-rest.php]: Fixed

File: plugin/webpay-rest.php (hunk 2 above)
Comment (mastudillot) [reply plugin/webpay-rest.php]: Fixed

File: plugin/views/admin/components/notice-missing-woocommerce.php (hunk 1 above)
Comment (mastudillot) [reply plugin/views/admin/components/notice-missing-woocommerce.php]: Fixed

**Review by Matiasnickolas — APPROVED** [review]

**Pull request merged**